publisher-sdk/src/test/java/com/criteo/publisher/model/CdbRequestFactoryTest.kt

```diff
@@ -100,9 +100,9 @@ class CdbRequestFactoryTest {
 
     val expectedSlot = CdbRequestSlot.create(
         "impId",
-        adUnit.placementId,
+        adUnit.placementId!!,
```

Comment:
It depends on whether ```CacheAdUnit#size``` and ```CacheAdUnit#placementId``` can be non-null. ```size``` cannot be non-null as interstitials don't have one. May be possible for placementId though?

Reply:
It's needed because `CacheAdUnit` members are nullable (size and adUnit), but `CdbRequestSlot#create` takes non-null params.

Reply:
Members are not explicitly @NonNull but they can be. CachedAdUnit are created in AdUnitMapper class. It validates the instance in `filterInvalidCacheAdUnits`.

And even the interstitial has a size as it takes the size of the screen.

Reply:
Ok, it's just that there were test cases explicitly covering the null scenarios so I thought it was a possibility. Can I remove all the test scenarios that pass null for AdUnit and size in this case? (eg: https://github.com/criteo/android-publisher-sdk/blob/main/publisher-sdk/src/test/java/com/criteo/publisher/model/AdUnitMapperTest.java#L86).

Reply:
These are tests checking the `filterInvalidCacheAdUnits` method.

To avoid confusion and to resume:
1. `AdUnit` (banner, interstitial, native) have nullable properties only to accommodate publishers
2. `CachedAdUnit` have non nullable properties and this is currently checked by `filterInvalidCacheAdUnits` rather than by compiler/annotation+linter.

As your working with `CachedAdUnit` here, you don't need the `!!` operator then.

This is just FAR, but I guess that for 1. we can ask ourselves if we want to keep this lenient API for v4. And for 2., we might want that the compiler does the job rather than a method.

Reply:
Let's sync over Zoom on this later.